Remove gap between sidebar boxes (for POIs)

JoeyStk opened the pull request:
### Short description
With introducing the collapsible boxes to the page, event and POI form we created a gap between the boxes when certain boxes are collapsed. This is mainly due to two factors: a. using grid instead of flex, b. the two sidebars for medium and 4xl screens aren't wrapped inside one parent div. These changes are going to be relevant for all post types, however in this PR (at least up until now) I focused on POIs only. If we fixed everything for POI I will extend these changes to events and pages one by one.

For a graphical summary on what I tried to fix see the screenshots here: 
Before: 
![grafik](https://user-images.githubusercontent.com/72705147/215739220-9de88b88-74f1-493b-8eae-21cdadc15322.png)

After: 
![grafik](https://user-images.githubusercontent.com/72705147/215739581-325686dd-1697-4098-b505-16aad08d6f24.png)


### Proposed changes

- Using grid only for 3xl screens
- Add parent div for two sidebars (for medium and 4xl screens)
- Remove flex-auto for the editor so it doesn't create blank space
- Adjust spacing between the boxes


### Side effects
While testing this, I didn't run into any. However I can't imagine that there are now, since this is a major change. So I would ask you to test this PR thoroughly :blush: 
Please also check if you can find classes, that aren't used anymore, so I can remove them :) 


### Resolved issues
Fixes: Parts of #2032 


__________________________________________________
<!-- Keep this link for the potential reviewer -->
[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

MizukiTemma reviewed (approved):
Thank you for PR :+1 Looks very nice 😃 

Do we need an entry in changelog?

seluianova reviewed (changes requested):
No gap now ✅
But there is a regression change:
If I collapse some blocks and re-open form, blocks are reordered randomly.
![image](https://user-images.githubusercontent.com/115008338/216973695-e47ca752-d102-4a80-8b14-4b059df44a34.png)
It's not reproduced on develop branch, all blocks always retain their position.
Do we want and can avoid it?

JoeyStk commented:
Thank you for your reviews!

@MizukiTemma I will add an entry to the changelog once I have extended this to events and pages :)
@selulanova This is expected behaviour, that was part of [this issue](https://github.com/digitalfabrik/integreat-cms/issues/1769). However since I created the PR we have added the option to switch off the automatic distribution of the boxes. I have rebased my branch now, which means the boxes shouldn't be automatically reorganized by default, but you could enable it in the user settings.

seluianova reviewed (commented):
Thank you 👌
One more regression issue I have noticed:
If I minimize the window, boxes go down and there is no gap between left and right columns.
![image](https://user-images.githubusercontent.com/115008338/217270279-a5f37809-0783-4a79-a21c-a41caf2ea4c9.png)

JoeyStk commented:
Oh, good catch. I don't know why I didn't see this. The gap-2 is of course not working anymore. I should have thought of that. I think it's fixed now :)

timoludwig reviewed (changes requested):
Hmm, somehow I noticed problems on all tested screen resolutions...
In my opinion, the spacing between the boxes and the columns should be consistent. I think, 4rem looks better than 2rem, but the most important thing is that it's consistent.

<details>
  <summary>1700px and below</summary>

![Screenshot 2023-02-10 at 13-16-16 Integreat Editorial System](https://user-images.githubusercontent.com/16021269/218091986-fab3b6bc-885f-4de1-aaf2-c69139fd9a22.png)

</details>

<details>
  <summary>1700px - 2100px</summary>

![Screenshot 2023-02-10 at 13-16-40 Integreat Editorial System](https://user-images.githubusercontent.com/16021269/218092498-08494e73-4aaf-42cc-a33a-4189ece31a7c.png)


</details>

<details>
  <summary>2100px and above</summary>

![Screenshot 2023-02-10 at 13-15-32 Integreat Editorial System](https://user-images.githubusercontent.com/16021269/218092516-fefb4e2b-5ad5-4c1b-8d04-b92aa433424c.png)


</details>

JoeyStk commented:
Very good catch. Thank you for the detailed feedback. In some parts this goes beyond what I wanted to do with this PR and in other parts I created this issue during a rebase. I adjusted the spacing for all boxes and gaps to class category 4 (e.g. gap-4 or mb-4, =1rem). I was only able to double check if my changes work for screens smaller than 2100px. For screens above 2100px I can only test it earliest next week

seluianova reviewed (approved):
Looks good to me!

timoludwig commented on CHANGELOG.md:
```diff
@@ -1,6 +1,8 @@
 UNRELEASED
 ----------
 
+* [ [#2032](https://github.com/digitalfabrik/integreat-cms/issues/2032) ] Remove gap between sidebar boxes (for POIs)fields to POI
```
```suggestion
* [ [#2032](https://github.com/digitalfabrik/integreat-cms/issues/2032) ] Remove gap between sidebar boxes in location form
```

timoludwig reviewed (changes requested):
Thanks! :+1:

Now I cannot find anymore problems, looks good!
I would just reword the changelog entry.
